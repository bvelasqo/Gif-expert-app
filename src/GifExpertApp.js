import React from "react";
import {AddCategory} from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

export const GifExpertApp = () => {
	const [categories, setCategories] = React.useState(["Dragon Ball"]);

	return (
		<>
			<h1>Gif Expert App</h1>
			<AddCategory setCategories={setCategories} />
			<hr />
			<ol>
				{
					// categories is an array of strings (categories)
					categories.map((category, index) => (
						//key is important to avoid bugs, never use index as key in arrays or objects (it's not unique)
						//category is the value of the array item (string)
						<GifGrid key={category} category={category} />
					))
				}
			</ol>
		</>
	);
};
