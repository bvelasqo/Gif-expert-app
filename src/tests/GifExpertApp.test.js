import Shallow from "enzyme/Shallow";
import {GifExpertApp} from "../GifExpertApp";

describe("GifExpertApp", () => {
  let categories = ["Dragon Ball", "Naruto"];
	let wrapper = Shallow(<GifExpertApp />);
	beforeEach(() => {
    categories = ["Dragon Ball", "Naruto"];
		wrapper = Shallow(<GifExpertApp defaultCategories={categories} />);
	});

	test("should render correctly", () => {
		expect(wrapper).toMatchSnapshot();
	});

  test("Should to show a list of categories", () => {
		expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
