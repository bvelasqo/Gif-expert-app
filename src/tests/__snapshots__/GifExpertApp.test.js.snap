// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp Should to show a list of categories 1`] = `
<Fragment>
  <h1>
    Gif Expert App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
  </ol>
</Fragment>
`;

exports[`GifExpertApp should render correctly 1`] = `
<Fragment>
  <h1>
    Gif Expert App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
  </ol>
</Fragment>
`;
